controllers/dataprotection/backup_controller.go

```diff
@@ -417,27 +427,80 @@ func (r *BackupReconciler) doNewPhaseAction(
 	return intctrlutil.Reconciled()
 }
 
-// handlePersistentVolumeClaim handles the persistent volume claim for the backup, the rules are as follows
+func (r *BackupReconciler) handlePersistentVolumeClaim(reqCtx intctrlutil.RequestCtx,
+	backup *dataprotectionv1alpha1.Backup,
+	backupPolicyName string,
+	commonPolicy *dataprotectionv1alpha1.CommonBackupPolicy) (pvcName string, pvName string, err error) {
+	// check the legacy PVC field first for compatibility,
+	// and fallback to the backup repo approach if the legacy PVC field is empty
+	if commonPolicy.PersistentVolumeClaim.Name != nil {
+		pvcName = *commonPolicy.PersistentVolumeClaim.Name
+	}
+	pvName, err = r.handlePersistentVolumeClaimLegacy(reqCtx, backup.Spec.BackupType, backupPolicyName, commonPolicy)
+	if err == nil || !intctrlutil.IsTargetError(err, intctrlutil.ErrorTypeBackupPVCNameIsEmpty) {
+		return pvcName, pvName, err
+	}
+
+	// check the PVC from backup repo
+	repo, err := r.getBackupRepo(reqCtx, backup, commonPolicy)
+	if err != nil {
+		return "", "", err
+	}
+	pvcName = repo.Status.BackupPVCName
+	if pvcName == "" {
+		err = intctrlutil.NewBackupPVCNameIsEmpty(string(backup.Spec.BackupType), backupPolicyName)
+		return "", "", err
+	}
+	pvc := &corev1.PersistentVolumeClaim{}
+	err = r.Client.Get(reqCtx.Ctx, client.ObjectKey{
+		Namespace: reqCtx.Req.Namespace,
+		Name:      pvcName,
+	}, pvc)
+	if err != nil && !apierrors.IsNotFound(err) {
+		// error occurred
+		return "", "", err
+	}
+	if err == nil {
+		// the PVC is already present
+		_, err = r.patchBackupObjectLabels(reqCtx, backup, map[string]string{
+			dataProtectionBackupRepoKey: repo.Name,
+		})
+		return pvcName, pvc.Spec.VolumeName, err
+	}
+	// the PVC is not present
+	// add a special annotation and wait for the backup repo controller
+	// to create the PVC.
+	_, err = r.patchBackupObjectLabels(reqCtx, backup, map[string]string{
+		dataProtectionBackupRepoKey:  repo.Name,
+		dataProtectionNeedRepoPVCKey: trueVal,
+	})
```

Comment:
if the backupRepo is empty in policy.spec, should set the default repo to policy.spec at this moment.
prevent changes of default repo.

Reply:
Label is a simple constraint, referencing PVC will backfill the default storageClass to spec

Reply:
As discussed offline, this part will remain unchanged for now.